From 164daa56d2d52cf55f9a4ab2d0308bc9834d7dab Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Wed, 10 Dec 2014 10:38:11 +0100
Subject: [PATCH] Fix logical expression in eap_client_active macro

Currently the expression always evaluates to true and gcc gives following
warning:

> eap.c:236:2: warning: logical 'or' of collectively exhaustive tests is always
true [-Wlogical-op]

Resolves: #1023620
---
 pppd/eap.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pppd/eap.h b/pppd/eap.h
index 3fa5391..087baad 100644
--- a/pppd/eap.h
+++ b/pppd/eap.h
@@ -110,8 +110,8 @@ enum eap_state_code {
 	"SRP1", "SRP2", "SRP3", "MD5Chall", "Open", "SRP4", "BadAuth"
 
 #ifdef USE_EAPTLS
-#define	eap_client_active(esp)	((esp)->es_client.ea_state != eapInitial ||\
-				 (esp)->es_client.ea_state != eapPending ||\
+#define	eap_client_active(esp)	((esp)->es_client.ea_state != eapInitial &&\
+				 (esp)->es_client.ea_state != eapPending &&\
 				 (esp)->es_client.ea_state != eapClosed)
 #else
 #define eap_client_active(esp)	((esp)->es_client.ea_state == eapListen)
-- 
1.8.3.1

